lib/wiki_assignment_output.rb

```diff
@@ -0,0 +1,97 @@
+require './lib/wiki'
+#= Class for generating wikitext for updating assignment details on talk pages
+class WikiAssignmentOutput
+  ################
+  # Entry points #
+  ################
+  def self.build_talk_page_update(title,
+                                  talk_title,
+                                  title_assignments,
+                                  course_page)
+    initial_page_content = Wiki.get_page_content talk_title
+    # We only want to add assignment tags to non-existant talk pages if the
+    # article page actually exists. This also servces to make sure that we
+    # only post to talk pages of mainspace articles, as we assume that pages
+    # like Talk:Template:Citation or Talk:Wikipedia:Notability do not exist.
+    if initial_page_content.nil?
+      return nil if Wiki.get_page_content(title).nil?
+      initial_page_content = ''
+    end
+
+    # Limit it to live assignments for this article-course
+    title_assignments = title_assignments.select { |a| !a['deleted'] }
+
+    course_assignments_tag = assignments_tag(course_page, title_assignments)
+    page_content = build_assignment_page_content(course_assignments_tag,
+                                                 course_page,
+                                                 initial_page_content)
+    page_content
+  end
+
+  ###################
+  # Helper methods #
+  ###################
+  def self.assignments_tag(course_page, title_assignments)
+    return '' if title_assignments.empty?
+
+    # Make a list of the assignees, role 0
+    tag_assigned = build_wikitext_user_list(title_assignments, 0)
```

Comment:
This makes total sense. (We have magic numbers in many places.)

What's the best place to store such magic numbers? Should these be global constants, or defined within the class?